versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/AbstractMergeTransplant.java

```diff
@@ -148,14 +168,15 @@ void apply(
         throws Exception;
   }
 
-  private Hash[] mergeTransplant(MergeOrTransplant mergeOrTransplant) throws Exception {
+  private Hash[] mergeTransplant(int numCommits, MergeOrTransplant mergeOrTransplant)
+      throws Exception {
     BranchName main = BranchName.of("main");
     BranchName branch = BranchName.of("branch");
     BranchName conflict = BranchName.of("conflict");
 
     databaseAdapter.create(branch, databaseAdapter.hashOnReference(main, Optional.empty()));
 
-    Hash[] commits = new Hash[3];
+    Hash[] commits = new Hash[numCommits];
```

Comment:
I remember that my PR for same issue got blocked because there was no strict validation. But this PR also don't have any validation but went ahead. 
I am not sure what is the process here.

Reply:
Your change was complex and changed the algorithm/implementation to generate the key-lists.
Dmitri's change did not change that algorithm.

Reply:
I also checked code coverage in IntelliJ (manually) - all branches in the new `if` statements were covered :)